cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1278,14 +1278,14 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
                     .compose(sts -> {
                         Storage oldStorage = getOldStorage(sts);
 
-                        this.zkCluster = ZookeeperCluster.fromCrd(kafkaAssembly, versions, oldStorage);
-                        this.zkService = zkCluster.generateService();
-                        this.zkHeadlessService = zkCluster.generateHeadlessService();
-
                         if (sts != null && sts.getSpec() != null)   {
                             this.zkCurrentReplicas = sts.getSpec().getReplicas();
                         }
 
+                        this.zkCluster = ZookeeperCluster.fromCrd(kafkaAssembly, versions, oldStorage, zkCurrentReplicas != null ? zkCurrentReplicas : 0);
```

Comment:
Maybe not related to this PR but why the `zkCurrentReplicas` is declared as `Integer` and not just a simple `int` which makes a little bit different the way we pass the old replicas between ZooKeeper and Kafka in the next code? Can the `getReplicas()` return `null` on a StatefulSet? The same is used in the `KafkaClusterDescription` but setting an `int` so I would say no.

Reply:
`zkCurrentReplicas` is used by the Zookeeper scaling where `null` indicates bascially new cluster. Therefore it exists as Integer and is a class variable. Thsi is there for a long time, I really just reuse the variable instead of creating another local one just for the Storage diff. 

For Kafka we do not use it anywhere for anything else. So I created a new local variable and since the storage diff has a different requirements it is sufficient this time to use `int`.

Reply:
Ah ok, now I see `zkCurrentReplicas` stays `null` when there is no STS yet, so new cluster. Thanks.